tools/cli/commands-packages.js

```diff
@@ -1192,31 +1203,101 @@ main.registerCommand({
 
       const expandedAlready = (deps.length > 0 && dontExpand.has(packageName));
       const shouldExpand = (deps.length > 0 && !expandedAlready && !isWeak);
-      if (indent1 !== '') {
-        indent1 += (shouldExpand ? '┬' : '─') + ' ';
+
+      // with normal tree display we send the current info to stdout
+      if (showTree) {
+        if (indent1 !== '') {
+          indent1 += (shouldExpand ? '┬' : '─') + ' ';
+        }
+
+        let suffix = (isWeak ? '[weak]' : '');
+        if (expandedAlready) {
+          suffix += topLevelSet.has(packageName) ? ' (top level)' : ' (expanded above)';
+        }
+
+        Console.info(indent1 + packageName + '@' + packageToPrint.version + suffix);
       }
 
-      let suffix = (isWeak ? '[weak]' : '');
-      if (expandedAlready) {
-        suffix += topLevelSet.has(packageName) ? ' (top level)' : ' (expanded above)';
+      // with json we add detailed info to the json object
+      if (showJson) {
+        if (expandedAlready) {
+          // on expanded packages we only want to add minimal information to
+          // keep the json file compact, so we make the value a stirng
+          if (topLevelSet.has(packageName)) {
+            parent[packageName] = `${packageToPrint.version}-(top-level)`
+          } else {
+            parent[packageName] = `${packageToPrint.version}-(expanded-above)`
+          }
+        } else {
+          // on non-expanded packages we want detailed information but we
+          // omit falsy values in order to keep the output minimal and readable
+          const entry = {};
+          parent[packageName] = entry;
+
+          const mapInfo = projectContext.packageMap.getInfo(packageName);
+
+          Object.entries({
+            version: packageToPrint.version,
+            local: mapInfo && mapInfo.kind === 'local',
+            weak: isWeak,
+            newerVersion: getNewerVersion(packageName, packageToPrint.version, catalog.official),
+            earliestCompatibleVersion: showDetails && packageToPrint.earliestCompatibleVersion,
```

Comment:
you could write this as:

...(showDetails ? packageToPrint : {})

Reply:
Resolved in 8937648